src/dex-ui/pages/dao/MultiSigDAODashboard/TransactionsList.tsx

```diff
@@ -13,12 +13,28 @@ const transactionTabs = [
 ];
 
 export function TransactionsList() {
-  const { dao } = useOutletContext<MultiSigDAODetailsContext>();
-  const { accountId: daoAccountId, safeId: safeAccountId } = dao;
+  const { dao } = useOutletContext<DAODetailsContext>();
+  const { accountId: daoAccountId } = dao;
+  let safeAccountId = "";
+  let tokenId = "";
+  let governanceAddress = "";
+  if (dao.type === DAOType.MultiSig) {
+    safeAccountId = dao.safeId;
+  } else if (dao.type === DAOType.GovernanceToken) {
+    tokenId = dao.tokenId;
+    governanceAddress = dao.governanceAddress;
+  }
   const { tabIndex, handleTabChange } = useTabFilters();
   const transactionFilters = transactionTabFilters.at(tabIndex) ?? defaultTransactionFilters;
 
-  const daoTransactionsQueryResults = useDAOProposals(daoAccountId, dao.type, safeAccountId, transactionFilters);
+  const daoTransactionsQueryResults = useDAOProposals(
+    daoAccountId,
+    dao.type,
+    safeAccountId,
+    transactionFilters,
+    governanceAddress,
+    tokenId
+  );
   const { isSuccess, isLoading, isError, error, data: transactions } = daoTransactionsQueryResults;
 
```

Comment:
For the future improvements we could use separate hooks for all DAO's proposals this would make debugging easy and hooks can be controlled using `enabled` property which can be passed as param.

Reply:
Added a TODO here for the time. will pick it up after finishing proposal details flow.